sdk/core/core-rest-pipeline/src/policies/bearerTokenAuthenticationPolicy.ts

```diff
@@ -219,3 +317,74 @@ export function bearerTokenAuthenticationPolicy(
     },
   };
 }
+
+/**
+ *
+ * Interface to represent a parsed challenge.
+ *
+ * @internal
+ */
+interface AuthChallenge {
+  scheme: string;
+  params: Record<string, string>;
+}
+
+/**
+ * Converts: `Bearer a="b", c="d", Pop e="f", g="h"`.
+ * Into: `[ { scheme: 'Bearer', params: { a: 'b', c: 'd' } }, { scheme: 'Pop', params: { e: 'f', g: 'h' } } ]`.
+ *
+ * @internal
+ */
+export function parseChallenges(challenges: string): AuthChallenge[] {
+  // Split the input string by schemes and parameters
+  // Challenge regex seperates the string to individual challenges with different schemes in the format `Scheme a="b", c="d"`
+  const challengeRegex = /(\w+) ((?:\w+="[^"]*",?\s*)+)/g;
+  // Parameter regex captures the claims group removed from the scheme in the format `a="b"` and `c="d"`
+  const paramRegex = /(\w+)="([^"]*)"/g;
```

Comment:
What if a parameter value is not surrounded by quotes? In this example the `algorithm` parameter does not have quotes: https://datatracker.ietf.org/doc/html/rfc7616#section-3.9.1. Do we still parse out the fields we care about correctly if something like that is present?

Reply:
@timovv  Seems like it doesn't capture if it's not surrounded by quotes.
 `(\w+)=(.+)` How about this? But this leaves out the WWW-Authenticate header which has `:` instead of `=`

Reply:
https://regex101.com/r/BAYzvq/1

Reply:
That doesn't work since it relies on there being a newline between each parameter. If I remove the newlines then it doesn't work anymore: https://regex101.com/r/iFrG1g/1. In practice the challenge won't have new lines in it so it's an issue.

The existing pattern is fine, so long as
1. we know that any time we get a CAE challenge the parameters we care about ("claims" and "error") will always have quotes, even if other parameters don't, and
1. other parameters not having quotes doesn't affect our ability to parse the parameters we need to for CAE (we can add a test for this)

I just want us to confirm those things.

Reply:
I can double check with the Identity crew. Based on the PRs I've seen, there seems to be an assumption that we only care about the params in quotes, and CAE claims & error params should be formatted as such

Reply:
I updated so that the challenge RegEx will capture params with or without quotes so it won't affect the parameter parsing and added a test for it. For the params RegEx, we only care about parameter in quotes so it's still kept the same

Reply:
we could add a comment above the params regex saying that for cae challenge, the challenge parameters are always quoted along with the link Charles referenced. @minhanh-phan

Reply:
@KarishmaGhiya updated in latest commit